torch/csrc/api/include/torch/nn/module.h

```diff
@@ -37,30 +80,83 @@ class Module {
   virtual ~Module() = default;
 
   /// Returns the name of the `Module`.
+  ///
+  /// A `Module` has an associated `name`, which is a string representation of
+  /// the kind of concrete `Module` it represents, such as `"Linear"` for the
+  /// `Linear` module. Under most circumstances, this name is automatically
+  /// inferred via runtime type information (RTTI). In the unusual circumstance
+  /// that you have this feature disabled, you may want to manually name your
+  /// `Module`s by passing the string name to the `Module` base class'
+  /// constructor.
   const std::string& name() const noexcept;
 
   /// Performs a recursive deep copy of the module and all its registered
-  /// parameters, buffers and submodules, optionally setting the current device
+  /// parameters, buffers and submodules.
+  ///
+  /// Optionally, this method sets the current device
   /// to the one supplied before cloning. If no device is given, each
   /// parameter and buffer will be moved to the device of its source.
+  ///
+  /// \rst
+  /// .. attention::
+  ///   Attempting to call the `clone()` method inherited from the base `Module`
+  ///   class (the one documented here) will fail. To inherit an actual
+  ///   implementation of `clone()`, you must subclass `Cloneable`. `Cloneable`
+  ///   is templatized on the concrete module type, and can thus properly copy a
+  ///   `Module`. This method is provided on the base class' API solely for an
+  ///   easier-to-use polymorphic interface.
+  /// \endrst
   virtual std::shared_ptr<Module> clone(
       at::optional<Device> device = at::nullopt) const;
 
   /// Provides a means to traverse the `Module` tree.
+  ///
+  /// See the documentation for `CursorBase` for information on how to operate
+  /// on the returned cursor.
   ModuleCursor modules();
+
+  /// Provides a means to traverse the `Module` tree.
+  ///
+  /// See the documentation for `CursorBase` for information on how to operate
```

Comment:
OOC: is it necessary to backtick class names? I was reading the Doxygen format manual, and it seemed to claim that ANY camelcase identifier would get hyperlinked if there was class of the same name.

Reply:
Things do get hyperlinked automatically, I just have a tendency to backtick any code things I mention. Maybe not necessary indeed